scalafix-reflect/src/main/scala-3/scalafix/internal/reflect/RuleCompiler.scala

```diff
@@ -0,0 +1,81 @@
+package scalafix.internal.reflect
+
+import dotty.tools.dotc.Compiler
+import dotty.tools.dotc.Run
+import dotty.tools.dotc.core.Contexts.FreshContext
+import dotty.tools.dotc.util.SourceFile
+import dotty.tools.dotc.reporting.StoreReporter
+import dotty.tools.dotc.interactive.InteractiveDriver
+import dotty.tools.io.AbstractFile as DottyAbstractFile
+
+import scala.reflect.io.VirtualDirectory
+import scala.reflect.io.AbstractFile
+import scala.reflect.internal.util.AbstractFileClassLoader
+
+import metaconfig.Configured
+import metaconfig.Input
+import metaconfig.ConfError
+import metaconfig.Position
+
+class RuleCompiler(
+    classpath: String,
+    target: AbstractFile = new VirtualDirectory("(memory)", None)
+) {
+  // we are not configuring any settings!
+  // private val settings = new Settings()
+  // settings.deprecation.value = true // enable detailed deprecation warnings
+  // settings.unchecked.value = true // enable detailed unchecked warnings
+  // settings.outputDirs.setSingleOutput(target)
+  // settings.classpath.value = classpath
+  private val settings = "-unchecked" :: "-deprecation" :: Nil
```

Comment:
I am not familiar with compiler options but aren't these making compilation stricter? Not sure what was the original intent in the Scala 2 counterpart, but in the context of Scala 3 I think we should be as permissive as possible.

Reply:
In this case the two options don't make the compilation stricter, they just print out additional warnings, from the man page:
```
-deprecation  Emit warning and location for usages of deprecated APIs.

-unchecked  Enable additional warnings where generated code depends on assumptions.
```
It could also be that we don't really want the additional warnings in the beginning.

Reply:
if it's just warnings let's keep them for now :+1: